gateway/src/shard/processor/impl.rs

```diff
@@ -332,6 +336,14 @@ impl ShardProcessor {
                     self.listeners.remove_all();
                     return;
                 }
+                Err(ReceivingEventError::SocketForwarder) => {
+                    tracing::warn!(
+                        "socket forwarder is closed, reconnecting"
+                    );
+
+                    self.reconnect(true).await;
```

Comment:
This probably doesn't need to be a full reconnect, we can try a resume first (the `resume` method will do that which is applicable to try first)